app/response_processor.py

```diff
@@ -84,50 +110,37 @@ def _send_receipt(self, decrypted, xml):
-            res = session.post(endpoint, data=xml, headers=headers, verify=False, auth=auth)
-
-            res_logger = res_logger.bind(status=res.status_code)
-
-            if res.status_code == 400:
-                res_logger.error("Receipt rejected by endpoint")
-                raise QuarantinableError
-
-            elif res.status_code == 404:
-                namespaces = {'error': 'http://ns.ons.gov.uk/namespaces/resources/error'}
-                tree = etree.fromstring(res.content)
-                element = tree.find('error:message', namespaces).text
-                elements = element.split('-')
-
-                if elements[0] == '1009':
-                    stat_unit_id = elements[-1].split('statistical_unit_id: ')[-1].split()[0]
-                    collection_exercise_sid = elements[-1].split('collection_exercise_sid: ')[-1].split()[0]
-
-                    res_logger.error("Receipt rejected by endpoint",
-                                     msg="No records were found on the man_ce_sample_map table",
-                                     error=1009,
-                                     stat_unit_id=stat_unit_id,
-                                     collection_exercise_sid=collection_exercise_sid)
-
+            self.logger.info("Calling external receipting service", service="External receipt")
+            session = Session()
+            retries = Retry(total=5, backoff_factor=0.5)
+            session.mount('http://', HTTPAdapter(max_retries=retries))
+            session.mount('https://', HTTPAdapter(max_retries=retries))
+
+            response = session.post(endpoint, data=xml, headers=headers, verify=False, auth=auth)
+            self.logger = self.logger.bind(status=response.status_code)
+
+            try:
+                response.raise_for_status()
+                self.logger.info("Sent receipt")
+                return response
+            except HTTPError:
+                if response.status_code == 400:
+                    self.logger.error("Receipt rejected by endpoint")
                     raise QuarantinableError
-
+                elif response.status_code == 404:
+                    self._check_namespace_error(response)
                 else:
-                    res_logger.error("Bad response from endpoint")
+                    self.logger.error("Bad response from endpoint")
                     raise RetryableError
-
-            elif res.status_code != 200 and res.status_code != 201:
-                # Endpoint may be temporarily down
-                res_logger.error("Bad response from endpoint")
-                raise RetryableError
-
-            else:
-                res_logger.info("Sent receipt")
-
-            return res
-
         except MaxRetryError:
-            res_logger.error("Max retries exceeded (5) attempting to send to endpoint")
+            self.logger.error("Max retries exceeded (5) attempting to send to endpoint")
```

Comment:
The number of retries should probably be a constant so it won't get out of step between actual setting and log.

Should probably have the number of retries as a value too (e.g.
```
self.logger.errror("Max retries exceeded attempting to send message", max_retries=MAX_RETRIES)
```
)

Reply:
Agreed - I've addressed this by setting an instance variable `_retries` and accessing it for the log message and instantiating the session,